src/averageAges.js

```diff
@@ -37,7 +50,24 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filteredWomen = people.filter(person => {
+    const { sex, name } = person;
+
+    if (withChildren === undefined) {
+      return sex === 'f';
+    }
+
+    // * I add it, if will need to get info about women who don't have children,
+    // * if input `false` in `withChildren` parameter.
+    // * we have more flexible code, who have children or don't, and all women
+    return withChildren
+      ? sex === 'f' && people.some(personChild => personChild.mother === name)
+      : sex === 'f' && people.every(personChild => personChild.mother !== name);
```

Comment:
```suggestion
    return sex === 'f'  && 
      (withChildren
        ? people.some(personChild => personChild.mother === name)
        : true);
```
`people.every(personChild => personChild.mother !== name)` You don't need this logic. 
This function accept array of people, sort them by gender and return women average age. If `withChildren` not provided - we return average age of all women.